projects/packages/identity-crisis/src/class-identity-crisis.php

```diff
@@ -1221,4 +1223,68 @@ public static function get_mismatched_urls() {
 			'current_url' => $data['home'],
 		);
 	}
+
+	/**
+	 * Try to detect dynamic $_SERVER[] vars being used for WP_SITEURL or WP_HOME definitions by reading the wp-config.
+	 *
+	 * When a site URL is dynamic, it can lead to a Jetpack IDC. If dynamic usage is detected, helpful support info
+	 * will be shown on the IDC UI.
+	 *
+	 * The `jetpack_idc_dynamic_site_url_detected` transient is used to rate-limit this check to once per hour.
+	 *
+	 * @return bool True if dynamic site urls were detected in wp-config, false otherwise.
+	 */
+	public static function detect_dynamic_site_url() {
+		$transient_key = 'jetpack_idc_dynamic_site_url_detected';
+		$transient_val = get_transient( $transient_key );
+
+		if ( false !== $transient_val ) {
+			return (bool) $transient_val;
+		}
+
+		$path      = self::locate_wp_config();
+		$wp_config = $path ? file_get_contents( $path ) : false; // phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents
+		if ( $wp_config ) {
+			$matched = preg_match(
+				'/define ?\( ?[\'"](?:WP_SITEURL|WP_HOME).+(?:HTTP_HOST).+\);/',
```

Comment:
1. `HTTP_HOST` does not necessarily mean a dynamic URL. It might be static, and might be used for other reasons.
2. The line might be commented out, and we'll still detect it..
3. We don't take into account `SERVER_NAME`, which might also be dynamic. There's a myriad of ways to make this line dynamic we won't be able to detect without bringing in a lot of complexity, such us our very own `define( 'WP_SITEURL', DOCKER_REQUEST_URL );`

But I don't think fixing those issues would be worth the effort.
Instead of trying to fix all of the above doesn't really make sense, we could use softer wording in the notice. Possibly something like "Your site _might be_ using dynamic values..."

Reply:
Thanks for the thoughts Sergey, 1 & 2 definitely make sense. Looking at actual support issues, it's most commonly the `define('WP_SITEURL', 'https://' . $_SERVER['HTTP_HOST'] . '/'); ` causing problems.

I agree that more relaxed working should work here because it's we aren't taking any actions just providing information to the user.

Reply:
Did some renaming/commenting to clarify intent in https://github.com/Automattic/jetpack/pull/22805/commits/b7e5ebf5c084183bf3d38d1ed59bd9993d6ebfa3

And relaxed the phrasing as a start in https://github.com/Automattic/jetpack/pull/22805/commits/e948ad698d3a03400f2cabd7713c906d8d750dfc